dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/metric_config_panel_item.tsx

```diff
@@ -0,0 +1,233 @@
+
+export const MetricConfigPanelItem = ({ fieldOptionList, visualizations }: any) => {
+  const dispatch = useDispatch();
+  const { tabId } = useContext<any>(TabContext);
+  const explorerVisualizations = useSelector(selectExplorerVisualization)[tabId];
+  const { data } = visualizations;
+
+  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;
+
+  const initialConfigEntry: MetricListEntry = {
+    label: '',
+    aggregation: [],
+    custom_label: '',
+    name: '',
+  };
+
+  const [configList, setConfigList] = useState<MetricList>({
+    metrics: [initialConfigEntry],
+  });
+
+  useEffect(() => {
+    if (
+      data.rawVizData?.[visualizations.vis.name] &&
+      data.rawVizData?.[visualizations.vis.name].dataConfig
+    ) {
+      setConfigList({
+        ...data.rawVizData[visualizations.vis.name].dataConfig,
+      });
+    }
+  }, [
+    data.defaultAxes,
+    data.rawVizData?.[visualizations.vis.name]?.dataConfig,
+    visualizations.vis.name,
+  ]);
+
+  const updateList = (value: string, index: number, name: string, field: string) => {
+    const listItem = { ...configList[name][index], [field]: value };
+
+    const updatedList = {
+      ...configList,
+      [name]: [
+        ...configList[name].slice(0, index),
+        listItem,
+        ...configList[name].slice(index + 1, configList[name].length),
+      ],
+    };
+    setConfigList(updatedList);
+  };
+
+  const updateListWithAgg = (value: string[], index: number, name: string, field: string) => {
+    const updatedList = {
+      ...configList,
+      [name]: [
+        ...configList[name].slice(0, index),
+        {
+          ...configList[name][index],
+          [field]: [...value],
+        },
+        ...configList[name].slice(index + 1, configList[name].length),
+      ],
+    };
+    setConfigList(updatedList);
+  };
+
+  const handleServiceRemove = (index: number, name: string) => {
+    const arr = [...configList[name]];
+    arr.splice(index, 1);
+    const updatedList = { ...configList, [name]: arr };
+    setConfigList(updatedList);
+  };
+
+  const handleServiceAdd = (name: string) => {
+    const updatedList = { ...configList, [name]: [...configList[name], initialConfigEntry] };
+    setConfigList(updatedList);
+  };
+
+  const updateChart = () => {
+    dispatch(
+      renderExplorerVis({
+        tabId,
+        data: {
+          ...explorerVisualizations,
+          [visualizations.vis.name]: {
+            dataConfig: {
+              metrics: configList.metrics,
+            },
+          },
+        },
+      })
+    );
+  };
+
+  const getOptionsAvailable = (sectionName: string) => {
+    const selectedFields = {};
+    for (const key in configList) {
+      configList[key] && configList[key].forEach((field) => (selectedFields[field.label] = true));
+    }
+
+    return fieldOptionList
+      .filter((field) => !selectedFields[field.label])
+      .filter((field) => numericalTypes.includes(field.type));
```

Comment:
Instead of looping over the items multiple times, is it possible to do the check once? Something like:
`.filter((field) => !selectedFields[field.label] && numericalTypes.includes(field.type))`. Let me know.